### PR TITLE
SAMA-48 added support for video attribution

### PR DESCRIPTION
**Pull request by Oleksandr1414**
* preloader for emails has been updated
* now you can attach videos

also includes "SAMA-16 Implement uploading wheel during uploading images"

### CODE REVIEW

**Review by ccvlad — APPROVED** [review]
looks good!

**Review by velichko-artem — APPROVED** [review]

File: src/components/generic/messageComponents/MessageAttachment.js [hunk 1]
```diff
@@ -1,18 +1,100 @@
...
+        }}
+      />
+    );
+
+    //get first clip from video for bg
+
+    return video;
+  }, [url]);
+
+  const preloaderView = useMemo(() => {
```
Comment (DSuvoZera) [inline src/components/generic/messageComponents/MessageAttachment.js:38]: the same question here. Just wondering...

File: src/components/generic/messageComponents/MessageAttachment.js [hunk 2]
```diff
@@ -1,18 +1,100 @@
...
+  localUrl,
+  blurHash,
+}) {
+  const { hash } = useLocation();
+  const navigate = useNavigate();
+
+  const [loaded, setLoaded] = useState(false);
+  const videoRef = useRef(null);
+
+  const videoView = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (IgorKhomenko) [reply src/components/generic/messageComponents/MessageAttachment.js:19]: agree, we could extract it into separate component for better code readability

File: src/components/generic/messageComponents/MessageAttachment.js (hunk 1 above)
Comment (IgorKhomenko) [reply src/components/generic/messageComponents/MessageAttachment.js:38]: agree, we could extract it into separate component for better code readability

**Review by IgorKhomenko — APPROVED** [review]
approved with some comments

**Pull request merged**